bin/extract-ORFs.py

```diff
@@ -64,19 +86,59 @@ def translations(read):
         def translations(read):
             return read.translations()
 
+    if kozakInfo or kozakOnly:
+        if aa:
+            # Should the code break in this case?
+            print('Did you pass NA sequences? Kozak consensuses can only'
+                  'be found in a nucleotide sequence.', file=sys.stderr)
+        else:
+            from dark.dna import findKozakConsensus
+
+        def writeToKozakOut(kozakread, kozakInfoFile):
+            """
+            Writes out information about a Kozak sequence stored in kozakread
+            to the file name given in kozakInfoFile.
+            """
+            kozakout = open(kozakInfoFile, 'a')
+            kozakout.write('Read ID: ' + kozakread.id)
+            kozakout.write('\n')
+            kozakout.write('Kozak sequence: ' + kozakread.sequence)
+            kozakout.write('\n')
+            kozakout.write('Offset of the Kozak sequence A: ' +
+                           str(kozakread.start + 6))
+            kozakout.write('\n')
+            kozakout.write('Quality of the Kozak consensus: ' +
+                           str(kozakread.kozakQuality) + ' %')
+            kozakout.write('\n')
+            kozakout.write('\n')
+            kozakout.close()
+
     for read in reads:
         try:
             for translation in translations(read):
-                for orf in translation.ORFs():
-                    # Omit open ORFs if they're unwanted.
-                    if ((orf.openLeft or orf.openRight) and not allowOpenORFs):
-                        continue
-
+                for orf in translation.ORFs(allowOpenORFs):
                     # Check the length requirements, if any.
                     if ((minORFLength is None or len(orf) >= minORFLength) and
-                            (maxORFLength is None or
-                                len(orf) <= maxORFLength)):
-                        write(orf.toString('fasta'))
+                       (maxORFLength is None or len(orf) <= maxORFLength)):
+                        # If Kozak consensus information is wanted:
+                        if kozakInfo and not aa:
+                            kozakreads = list(findKozakConsensus(read))
+                            for kozakread in kozakreads:
+                                writeToKozakOut(kozakread, kozakInfoFile)
```

Comment:
It's not useful to take a generator, turn it into a list, then step through the list. That defeats the point of having a generator in the first place. So:

```python
for kozakread in findKozakConsensus(read):
    # Etc.
```

Or if you're feeling sinister:

```python
map(lambda kozakread: writeToKozakOut(kozakread, kozakInfoFile),
    findKozakConsensus(read))
```

Reply:
ah yes! That does indeed defeat the point... done (1st version)